src/common.rs

```diff
@@ -95,3 +95,8 @@ impl WriteConcern {
         bson
     }
 }
+
+pub fn merge_options<T: Into<bson::Document>>(document: bson::Document, options: T) -> bson::Document {
```

Comment:
Slight tangent: this method exemplifies so much of what I love about Rust. Bounded generics, composable standard library traits, a great iterator API, type inference; it's all here.